ibmcloud/image/build.sh

```diff
@@ -143,21 +162,52 @@ done < <(parted -s -m "$src_nbd" unit B print)
 sleep 1
 resize2fs -f "${src_nbd}p1"
 
-case "$disklabel" in
-    gpt)
-        sgdisk "$src_nbd" -R "$tmp_nbd"
-        sgdisk -G "$tmp_nbd"
-        ;;
-    msdos)
-        sfdisk -d "$src_nbd" | sfdisk "$tmp_nbd"
-        ;;
-esac
+if [ "${SE_BOOT}" = "1" ]; then
+    echo "Creating boot-se and root partitions" 
+    parted -a optimal $tmp_nbd mklabel gpt \
+        mkpart boot-se ext4 1MiB 256MiB \
+        mkpart root 256MiB "${disksize}" \
+        set 1 boot on
+else
+    case "$disklabel" in
+        gpt)
+            sgdisk "$src_nbd" -R "$tmp_nbd"
+            sgdisk -G "$tmp_nbd"
+            ;;
+        msdos)
+            sfdisk -d "$src_nbd" | sfdisk "$tmp_nbd"
+            ;;
+    esac
+fi
 
-for part_number in "${parts[@]}"; do
-    dd if="${src_nbd}p$part_number" of="${tmp_nbd}p$part_number" bs=$((1024*1024))
-done
+if [ "${SE_BOOT}" = "1" ]; then
+    echo "Waiting for the two nbd partitions to show up"
+    while true; do
+    sleep 1
+    [ -e ${tmp_nbd}p2 ] && break
+    done
+    printf "\nFormatting boot-se partition\n"
+    mke2fs -t ext4 -L boot-se ${tmp_nbd}p1
+    export BOOT_UUID=$(blkid ${tmp_nbd}p1 -s PARTUUID -o value)
+    export ROOT_UUID=$(blkid ${tmp_nbd}p2 -s PARTUUID -o value)
```

Comment:
I don't think we need to export `BOOT_UUID` and `ROOT_UUID`.

I prefer variable names with lower letters (`boot_uuid` and `root_uuid`) for internal variables.

Reply:
updated.